tokio/src/sync/tests/loom_watch.rs

```diff
@@ -34,3 +35,29 @@ fn smoke() {
         th.join().unwrap();
     })
 }
+
+#[test]
```

Comment:
@Darksonn @debadree25 In general, this test looks good to me. As a good practice I would suggest adding some comments to explain the purpose of each step in the test. 


```rust
#[test]
fn wait_for_test() {
    // Spawn threads with the loom runtime
    loom::model(|| {
        // Create a watch channel and clone the sender
        let (tx, mut rx) = watch::channel(false);
        let tx_arc = Arc::new(tx);
        let tx1 = tx_arc.clone();
        let tx2 = tx_arc.clone();

        // Spawn two threads to modify the value being watched
        let th1 = thread::spawn(move || {
            for _ in 0..3 {
                tx1.send_modify(|x| *x = true);
            }
        });

        let th2 = thread::spawn(move || {
            tx2.send(true).ok();
        });

        // Wait for the value to be set to true
        block_on(rx.wait_for(|x| *x)).unwrap();

        // Wait for the threads to finish
        th1.join().unwrap();
        th2.join().unwrap();
    });
}
```